flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/StateTable.java

```diff
@@ -166,6 +168,8 @@ public boolean isEmpty() {
 
 	public abstract Stream<K> getKeys(N namespace);
 
+	public abstract CloseableIterator<Tuple2<N, K>> getNamespaceKeyIterator();
```

Comment:
True, we can use state value provided by the iterator, instead of reading again, as the copy-on-write table iteration should deliver only actual up-to-date TTL values.